BubbleScan-AI/Bubble_Scan_AI.py

```diff
@@ -0,0 +1,413 @@
+import json
+import re
+import shutil
+import fitz  # PyMuPDF
+import os
+import cv2
+import numpy as np
+
+
+class Scantron95945:
+    def __init__(self, pdf_path):
+        self.pdf_name = None
+        self.pdf_path = pdf_path
+        self.source_folder = "data"
+        self.output_folder = "data"
+        self.template_path = "template.jpg"
+        self.extractImagesFromPdf()
+        self.template_matching()
+        self.extractROIs()
+        self.extract_responses()
```

Comment:
If extract_responses() returned json data, then the user of this class could create a Scantron95945 object and. then call extract_responses() to get the data. This would give a clean programmatic interface between Scantron95945 and whoever wants to use it.